### ISSUE TITLE
test case fixtures error

### ISSUE THREAD

**Issue by bitshine** [canni/YiiMongoDbSuite#21]
When use phpunit to test and using  fixtures, get an error:

1) FeedTest::testAddFeed
CException: EMongoDB does not have a method named "getSchema".

D:\www\yii-1.1.5.r2654\framework\base\CComponent.php:266
D:\www\yii-1.1.5.r2654\framework\test\CDbFixtureManager.php:231
D:\www\yii-1.1.5.r2654\framework\test\CDbFixtureManager.php:113
D:\www\yii-1.1.5.r2654\framework\test\CDbFixtureManager.php:85
D:\www\yii-1.1.5.r2654\framework\base\CModule.php:372
D:\www\yii-1.1.5.r2654\framework\test\CWebTestCase.php:69
D:\www\yii-1.1.5.r2654\framework\test\CWebTestCase.php:101
D:\www\feed\feed.cdeledu.com\protected\tests\WebTestCase.php:22

**Comment by canni** [comment]
You're using Yii default fixtures manager, change the fixtures manager to provided one in Yours config

**Comment by bitshine** [comment]
I already config in config/test.php

```
<?php
return CMap::mergeArray(
    require(dirname(__FILE__).'/main.php'),
    array(
        'components'=>array(
            'fixture'=>array(
                'class'=>'MongoDbFixtureManager',
            ),
            /* uncomment the following to provide test database connection
            'db'=>array(
                'connectionString'=>'DSN for test database',
            ),
            */
            'mongodb'=>array(
                'connectionString'=>'mongodb://localhost',
                'dbName'=>'feed_test',
                'class'=>'EMongoDB',
                'fsyncFlag'=>true,
                //'safe'=>true,
            ),
        ),
    )
);
```

 How can I config?

**Comment by canni** [comment]
Ah, I see now, You're using CWebTestCase, that is forcing standard Yii fixtures manager, I'll implement CWebTestCase for Mongo ext

**Comment by canni** [comment]
Hmm.. It should work, I've looked into CWebTestCase code, and it calls 'fixtures' component, do not create new.

So I think, You have not included proper test.php config file (ensure You have using index-test.php ? )

**Comment by pgaultier** [comment]
Hi, just to be sure, can you change : MongoDbFixtureManager to EMongoDbFixtureManager

I think you forget the starting E

**Comment by canni** [comment]
OK, will fix that :)

**Comment by canni** [comment]
Hmm... I dont see any fixture manager name without "E" ?

**Comment by pgaultier** [comment]
Well, as far as I can tell the problem is that the fixture manager is not correctly initialised : 

```
<?php
return CMap::mergeArray(
require(dirname(__FILE__).'/main.php'),
array(
    ...
    'components'=>array(
        'fixture'=>array(
            'class'=>'MongoDbFixtureManager',
        ),
    ...
    );
?>
```

should be replaced by 

```
<?php
return CMap::mergeArray(
require(dirname(__FILE__).'/main.php'),
array(
    ...
    'components'=>array(
        'fixture'=>array(
            'class'=>'EMongoDbFixtureManager',
        ),
    ...
    );
?>
```

And do not forget to make sure you have imported the directory *\* ext.YiiMongoDbSuite.test.\* **

**Comment by bitshine** [comment]
In the dir ext.YiiMongoDbSuite.test only one php file boostrap.php and a sub dir unit. No EMongoDbFixtureManager there.

**Comment by pgaultier** [comment]
This is because you are using the stable branch. The fixture manager is only in the devel branch right now.

check here : https://github.com/canni/YiiMongoDbSuite/tree/devel/test

Regards

**Comment by pgaultier** [comment]
As far as I can see, fixture manager should be available in 1.3.6
